bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/MyqBinding.java

```diff
@@ -0,0 +1,374 @@
+	 *            Configuration properties for this component obtained from the
+	 *            ConfigAdmin service
+	 */
+	public void activate(final BundleContext bundleContext,
+			final Map<String, Object> configuration) {
+		this.bundleContext = bundleContext;
+		modified(configuration);
+	}
+
+	/**
+	 * Called by the SCR when the configuration of a binding has been changed
+	 * through the ConfigAdmin service.
+	 * 
+	 * @param configuration
+	 *            Updated configuration properties
+	 */
+	public void modified(final Map<String, Object> configuration) {
+
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString)) {
+			refreshInterval = Long.parseLong(refreshIntervalString);
+		}
+
+		// update the internal configuration accordingly
+		String usernameString = (String) configuration.get("username");
+		String passwordString = (String) configuration.get("password");
+
+		// reinitialize connection object if username and password is changed
+		if (StringUtils.isNotBlank(usernameString)
+				&& StringUtils.isNotBlank(passwordString)) {
+			myqOnlineData = new MyqData(usernameString, passwordString);
+		}
+		invalidCredentials = false;
+		schedulePoll(refreshInterval);
+	}
+
+	/**
+	 * Called by the SCR to deactivate the component when either the
+	 * configuration is removed or mandatory references are no longer satisfied
+	 * or the component has simply been stopped.
+	 * 
+	 * @param reason
+	 *            Reason code for the deactivation:<br>
+	 *            <ul>
+	 *            <li>0 – Unspecified
+	 *            <li>1 – The component was disabled
+	 *            <li>2 – A reference became unsatisfied
+	 *            <li>3 – A configuration was changed
+	 *            <li>4 – A configuration was deleted
+	 *            <li>5 – The component was disposed
+	 *            <li>6 – The bundle was stopped
+	 *            </ul>
+	 */
+	public void deactivate(final int reason) {
+		this.bundleContext = null;
+		// deallocate resources here that are no longer needed and
+		// should be reset when activating this binding again
+
+		if (pollFuture != null && !pollFuture.isCancelled())
+			pollFuture.cancel(true);
+
+		if (pollResetFuture != null && !pollResetFuture.isCancelled())
+			pollResetFuture.cancel(true);
+	}
+
+	/**
+	 * Poll for device changes
+	 */
+	private void poll() {
+		if (invalidCredentials || this.myqOnlineData == null) {
+			logger.debug("Invalid Account Credentials");
+			return;
+		}
+
+		try {
+			// Get myQ Data
+			GarageDoorData garageStatus = myqOnlineData.getGarageData();
+
+			for (MyqBindingProvider provider : providers) {
+				for (String mygItemName : provider.getInBindingItemNames()) {
+					MyqBindingConfig deviceConfig = getConfigForItemName(mygItemName);
+
+					if (deviceConfig != null) {
+						GarageDoorDevice garageopener = garageStatus
+								.getDevice(deviceConfig.deviceIndex);
+						if (garageopener != null) {
```

Comment:
Recommend
```
State newState = UnDefType.UNDEF;
for (Class<? extends State> type : deviceConfig.acceptedDataTypes) {
  if (StringType.class == type) {
    // newState = new StringType...
    break;
  }
  else if (OpenClosedType.class == type) {
    // newState = OpenClosedType...
    break;
  }
  else {
    // etc...
  }
}
eventPublisher.postUpdate(mygItemName, newState);
```
this way you don't care about the kind of item, only the state types it supports.